addons/stock_landed_costs/models/account_move.py

```diff
@@ -53,23 +53,21 @@ class AccountMoveLine(models.Model):
     product_type = fields.Selection(related='product_id.detailed_type', readonly=True)
     is_landed_costs_line = fields.Boolean()
 
-    @api.onchange('is_landed_costs_line')
-    def _onchange_is_landed_costs_line(self):
-        """Mark an invoice line as a landed cost line and adapt `self.account_id`. The default
-        value can be set according to `self.product_id.landed_cost_ok`."""
-        if self.product_id:
-            accounts = self.product_id.product_tmpl_id._get_product_accounts()
-            if self.product_type != 'service':
-                self.account_id = accounts['expense']
-                self.is_landed_costs_line = False
-            elif self.is_landed_costs_line and self.move_id.company_id.anglo_saxon_accounting:
-                self.account_id = accounts['stock_input']
-            else:
-                self.account_id = accounts['expense']
+    @api.depends('is_landed_costs_line')
+    def _compute_account_id(self):
+        landed_costs_lines = self.filtered(lambda l: l.is_landed_costs_line and l.move_id.company_id.anglo_saxon_accounting)
+        for line in landed_costs_lines:
+            line.account_id = line.product_id.product_tmpl_id._get_product_accounts()['stock_input']
+        super(AccountMoveLine, self - landed_costs_lines)._compute_account_id()
 
     @api.onchange('product_id')
     def _onchange_is_landed_costs_line_product(self):
         if self.product_id.landed_cost_ok:
             self.is_landed_costs_line = True
         else:
             self.is_landed_costs_line = False
+
+    @api.onchange('is_landed_costs_line')
+    def _onchange_is_landed_costs_line_product(self):
```

Comment:
Two onchanges with the same name... :suspect: 
This onchange is useless, no ? It modifies the onchange's field and the other onchange seems to be enough I think

Reply:
Yeah, the first one was badly named IMO.
The second one is there to prevent that the field is wrongly set, and resets it. I tried to stay close to the previous behavior, by extracting the computation of the account into the compute method.

```python
    @api.onchange('is_landed_costs_line')
    def _onchange_is_landed_costs_line(self):
        """Mark an invoice line as a landed cost line and adapt `self.account_id`. The default
        value can be set according to `self.product_id.landed_cost_ok`."""
        if self.product_id:
            if self.product_type != 'service':
                self.is_landed_costs_line = False
```

Reply:
Since `product.product` `landed_cost_ok` already needs the product to be a service, I think if you add `is_landed_costs_line` in the first onchange's fields, it should be enough
```python
@api.onchange(`is_landed_costs_line`, 'product_id')
    def _onchange_product_id_landed_costs(self):
```
That said, if we can mark as a landed cost line an account move line with a service product who isn't `landed_cost_ok`, ignore my comment

Reply:
`landed_cost_ok`'s constraint on the product type it is implemented inside of the `write`, so pretty easy to bypass inside of the `create`...
This whole mechanism/module needs to be reviewed, but it is out of the scope of this PR and of my expertise...

_____________________
```python
@api.onchange(`is_landed_costs_line`, 'product_id')
    def _onchange_product_id_landed_costs(self):
```
Well that's the same a having a computed readonly field...
If I understood correctly, we can have a product that is marked as `landed_cost_ok` on a line that is not `is_landed_costs_line`, by unchecking the boolean box.
You solution doesn't allow that.